Flow.Launcher.Core/Plugin/PluginsLoader.cs

```diff
@@ -129,47 +133,104 @@ public static IEnumerable<PluginPair> PythonPlugins(List<PluginMetadata> source,
                 if (File.Exists(path))
                 {
                     Constant.PythonPath = path;
                 }
                 else
                 {
-                    Log.Error($"|PluginsLoader.PythonPlugins|Can't find python executable in {path}");
+                    Log.Error("PluginsLoader",$"Tried to automatically set from Settings.PythonDirectory " +
+                        $"but can't find python executable in {path}", "PythonPlugins");
                 }
             }
 
-            // if we have a path to the python executable,
-            // load every python plugin pair.
-            if (String.IsNullOrEmpty(Constant.PythonPath))
+            if (string.IsNullOrEmpty(settings.PythonDirectory))
             {
-                return new List<PluginPair>();
+                if (MessageBox.Show("Flow detected you have installed Python plugins, " +
+                        "would you like to install Python to run them? " +
+                        Environment.NewLine + Environment.NewLine +
+                        "Click no if it's already installed, " +
+                        "and you will be prompted to select the folder that contains the Python executable",
+                        string.Empty, MessageBoxButtons.YesNo) == DialogResult.No
+                    && string.IsNullOrEmpty(settings.PythonDirectory))
+                {
+                    var dlg = new FolderBrowserDialog
+                    {
+                        SelectedPath = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles)
+                    };
+
+                    var result = dlg.ShowDialog();
+                    if (result == DialogResult.OK)
+                    {
+                        string pythonDirectory = dlg.SelectedPath;
+                        if (!string.IsNullOrEmpty(pythonDirectory))
+                        {
+                            var pythonPath = Path.Combine(pythonDirectory, PythonExecutable);
+                            if (File.Exists(pythonPath))
+                            {
+                                settings.PythonDirectory = pythonDirectory;
+                                Constant.PythonPath = pythonPath;
+                            }
+                            else
+                            {
+                                MessageBox.Show("Can't find python in given directory");
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    var installation = Task.Run(async () => { await DroplexPackage.Drop(App.python3_9_1).ConfigureAwait(false); });
+                    installation.Wait();
+
+                    var installedPythonDirectory =
+                        Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), @"Programs\Python\Python39");
```

Comment:
What if user don't set the environment path when installing python?
BTW, isn't python installation require restart computer in order to setup the environmental variable?

Reply:
this is the env path for local app data folder, should always be set right

Reply:
Oh see it.